app/api/test_api/test_a_user_registration.py

```diff
@@ -1,122 +1,107 @@
-import  unittest
-from app.api.v1 import  application
+import unittest
+from app import create_app
 import json
+from ..v1.utils.error_file import en_errors
+
+application =create_app()
+
+data = ""
 
-data=""
 
 class TestUserLogin(unittest.TestCase):
 
     def setUp(self):
-
-        self.url_prefix=""
+        self.url_prefix = "api/v1"
         self.app = application.test_client()
 
     def test_a_signup_with_wrong_data(self):
         global data
-        response = self.app.post(self.url_prefix+"/signup",headers ={"Content-Type":"application/json"},data = "")
-        self.assertEqual(response.status_code,200,"signup method is allowed")
-        self.assertEqual(json.loads(response.get_data().decode()),{'error': 'this route required proper formated json '},"correct error message should be given back")
+        response = self.app.post(self.url_prefix + "/signup", headers={"Content-Type": "application/json"}, data="")
+        self.assertEqual(response.status_code, 400, "signup method is allowed")
+        self.assertEqual(json.loads(response.get_data().decode()),{'error': 'this route required proper formated json ', 'status': 400}
```

Comment:
Break this line. Have `json.loads()` on its line and assign a variable.

Reply:
thanks will certainly check it